Projects/Payment/Sources/Services/PaymentService/PaymentServiceOctopus.swift

```diff
@@ -11,7 +11,18 @@ public class hPaymentServiceOctopus: hPaymentService {
     public func getPaymentData() async throws -> PaymentData? {
         let query = OctopusGraphQL.PaymentDataQuery()
         let data = try await octopus.client.fetch(query: query, cachePolicy: .fetchIgnoringCacheCompletely)
-        return PaymentData(with: data)
+
+        let paymentDetailsQuery = OctopusGraphQL.PaymentInformationQuery()
+        let paymentDetailsData = try await octopus.client.fetch(
+            query: paymentDetailsQuery,
+            cachePolicy: .fetchIgnoringCacheCompletely
+        )
+
+        if paymentDetailsData.currentMember.paymentInformation.connection != nil {
+            let paymentDetails = PaymentData.PaymentDetails(with: paymentDetailsData)
+            return PaymentData(with: data, paymentDetails: paymentDetails)
+        }
+        return PaymentData(with: data, paymentDetails: nil)
```

Comment:
Instead of this:
```
        if paymentDetailsData.currentMember.paymentInformation.connection != nil {
            let paymentDetails = PaymentData.PaymentDetails(with: paymentDetailsData)
            return PaymentData(with: data, paymentDetails: paymentDetails)
        }
        return PaymentData(with: data, paymentDetails: nil)

```
we can do this
```
let paymentDetails = PaymentData.PaymentDetails(with: paymentDetailsData)
return PaymentData(with: data, paymentDetails: paymentDetails)
```

and instead of this
```
    init(with model: OctopusGraphQL.PaymentInformationQuery.Data) {
        self.paymentMethod = L10n.paymentsAutogiroLabel
        self.account = model.currentMember.paymentInformation.connection?.descriptor ?? ""
        self.bank = model.currentMember.paymentInformation.connection?.displayName ?? ""
    }
```

we can do this
```
    init?(with model: OctopusGraphQL.PaymentInformationQuery.Data) {
        guard let account =  model.currentMember.paymentInformation.connection?.descriptor,
              let bank = model.currentMember.paymentInformation.connection?.displayName else { return nil }
        self.paymentMethod = L10n.paymentsAutogiroLabel
        self.account = account
        self.bank = bank
    }
```
What do you think?
We are skipping having "return PaymentData(" 2 times and init PaymentDetails only when it is needed.

Reply:
yeah make sense